module/tasks/openchain.tasks.ps1

```diff
@@ -0,0 +1,156 @@
+# <copyright file="openchain.tasks.ps1" company="Endjin Limited">
+# Copyright (c) Endjin Limited. All rights reserved.
+# </copyright>
+
+# Synopsis: Stores the raw generated SBOM in an Azure storage account
+task PublishCovenantOutputToStorage `
+    -If { !$SkipBuildSolution -and $SolutionToBuild -and $PublishCovenantOutputToStorage } `
+    -After RunCovenant `
+    -Jobs {
+    if ( (Test-Path $covenantJsonOutputFile) -and `
+            $AnalysisOutputStorageAccountName -and `
+            $AnalysisOutputContainerName -and `
+            $AnalysisOutputBlobPath) {
+    
+        $covenantJsonOutputFilename = (Split-Path -Leaf $covenantJsonOutputFile)
+        $filename = "{0}-{1}.json" -f [IO.Path]::GetFileNameWithoutExtension($covenantJsonOutputFilename),
+                                     ([DateTime]::Now).ToString('yyyyMMddHHmmssfff')
+
+        Write-Information @"
+Publishing storage account:
+    Source File: $covenantJsonOutputFile
+    Account: $AnalysisOutputStorageAccountName
+    Blob Path: "$AnalysisOutputContainerName/$AnalysisOutputBlobPath/$filename"
+"@
+
+        $uri = "https://{0}.blob.core.windows.net/{1}/{2}/{3}" -f $AnalysisOutputStorageAccountName,
+                        $AnalysisOutputContainerName,
+                        $AnalysisOutputBlobPath,
+                        $filename
+
+        $authUri = & az storage blob generate-sas `

```

Comment:
Do we need to think about transient error handling and retries?

Reply:
I've added some basic retry behaviour